5-performance-optimization.ipynb

```diff
@@ -0,0 +1,996 @@
+{
+ "cells": [
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "f06a2f61-89b5-4a1f-9928-da4fe14d62a5",
+   "metadata": {},
+   "source": [
+    "<img src=\"images/dask_horizontal.svg\" width=\"45%\" alt=\"Dask logo\\\">\n",
+    "\n",
+    "# Optimizing Dask Workloads\n",
+    "\n",
+    "This notebook illustrates a common Dask ETL workload, and demonstrates how one might go about diagnosing and resolving performance issues using the dashboard. We will provide a motivating example for where a user may require Dask, and outline some common do's and don't's of Dask dataframe operations.\n",
+    "\n",
+    "## Cluster setup\n",
+    "\n",
+    "Because the data we are using resides in AWS S3, we will be spinning up a [Coiled cluster](https://docs.coiled.io/user_guide/clusters/index.html) in the same region to minimize I/O costs (Disclaimer: Some of the instructors for this tutorial are employed by Coiled.):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "29f3298c-6626-4eb4-8389-1e3ca7f83f67",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "\n",
+    "cluster = coiled.Cluster(n_workers=50, region=\"us-east-2\")  # start workers close to data to minimize costs\n",
+    "client = cluster.get_client()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "547788e0",
+   "metadata": {},
+   "source": [
+    "Once we have initialized a cluster and client, we can easily view the Dask dashboard either through widgets provided by [dask-labextension](https://github.com/dask/dask-labextension), or by visiting the dashboard URL directly:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "b3396b6a-6432-4e86-b826-17556efee8ec",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "client"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "id": "2a4c4623-e950-4685-849c-9793b5f18e59",
+   "metadata": {},
+   "source": [
+    "# Motivation: Uber/Lyft TLC Trip Records\n",
+    "\n",
+    "The New York City Taxi and Limousine Commission (TLC) collects trip record information for each taxi and for-hire vehicle trip completed by licensed drivers and vehicles; a subset of the [High-Volume For-Hire Services](https://www.nyc.gov/site/tlc/businesses/high-volume-for-hire-services.page) data (~60 GB) stored on S3 provides a good example of an out-of-core dataset that we would be unable to explore on a standard laptop due to memory limitations.\n",
```

Comment:
```suggestion
    "The New York City Taxi and Limousine Commission (TLC) collects trip record information for each taxi and for-hire vehicle trip completed by licensed drivers and vehicles; a subset of the [High-Volume For-Hire Services](https://www.nyc.gov/site/tlc/businesses/high-volume-for-hire-services.page) data (~60 GB) stored on S3 provides a good example of an out-of-core dataset that we would be hard to explore on a standard laptop due to memory limitations.\n",
```
I'm not sure if we would be "unable" but it would definitely be hard. I'm thinking that we for sure wouldn't be able to persist the dataset, but I do not think it is impossible, test it'll take a long time but not impossible. Thoughts?